git/repo/base.py

```diff
@@ -713,11 +713,14 @@ def blame_incremental(self, rev, file, **kwargs):
                            committed_date=int(props[b'committer-time']))
                 commits[hexsha] = c
             else:
-                # Discard the next line (it's a filename end tag)
-                line = next(stream)
-                tag, value = line.split(b' ', 1)
-                assert tag == b'filename', 'Unexpected git blame output'
-                orig_filename = value
+                # Discard all lines until we find "filename" which is
+                # guaranteed to be the last line
+                while True:
```

Comment:
Ah, I see now! I didn't know that `next` throws - iterators do that, right. Have been doing Rust for too long ;).